Source/SwiftLintFramework/Rules/FirebaseCoreRule.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  FirebaseCoreRule.swift
+//  SwiftLint
+//
+//  Created by Ibrahim Ulukaya (Google Inc.) on 3/8/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+
+public struct FirebaseCoreRule: ConfigurationProviderRule, OptInRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "firebase_core",
+        name: "Firebase Core",
+        description: "Firebase should be configured before use.",
+        nonTriggeringExamples: [
+            "class AppDelegate: UIResponder, UIApplicationDelegate {\n" +
+            "  func application(_ application: UIApplication," +
+            "      didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {\n" +
+            "    FirebaseApp.configure()\n" +
+            "    return true \n" +
+            "  }\n" +
+            "}"
+        ],
+        triggeringExamples: [
+            "class AppDelegate: UIResponder, UIApplicationDelegate {\n" +
+            "  func application(_ application: UIApplication," +
+            "      didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {\n" +
+            "    return true \n" +
+            "  }\n" +
+            "}"
```

Comment:
What if there's an object response for configuring all the 3rd-party software in the application. For instance: Firebase, Mixpanel, etc...